src/qibocal/protocols/characterization/ramsey.py

```diff
@@ -119,43 +120,72 @@ def _acquisition(
     # additionally include wait time and t_max
     data = RamseyData(params.n_osc, params.delay_between_pulses_end, detuning_sign=+1)
 
-    # sweep the parameter
-    for wait in waits:
-        for qubit in qubits:
-            RX90_pulses2[qubit].start = RX90_pulses1[qubit].finish + wait
-            ro_pulses[qubit].start = RX90_pulses2[qubit].finish
-            if params.n_osc != 0:
-                # FIXME: qblox will induce a positive detuning with minus sign
-                RX90_pulses2[qubit].relative_phase = (
-                    RX90_pulses2[qubit].start
-                    * data.detuning_sign
-                    * 2
-                    * np.pi
-                    * (params.n_osc)
-                    / params.delay_between_pulses_end
+    if params.n_osc != 0:
+        # sweep the parameter
+        for wait in waits:
+            for qubit in qubits:
+                RX90_pulses2[qubit].start = RX90_pulses1[qubit].finish + wait
+                ro_pulses[qubit].start = RX90_pulses2[qubit].finish
+                if params.n_osc != 0:
+                    RX90_pulses2[qubit].relative_phase = (
+                        RX90_pulses2[qubit].start
+                        * (-2 * np.pi)
+                        * (params.n_osc)
+                        / params.delay_between_pulses_end
+                    )
+
+            # execute the pulse sequence
+            results = platform.execute_pulse_sequence(
+                sequence,
+                ExecutionParameters(
+                    nshots=params.nshots,
+                    relaxation_time=params.relaxation_time,
+                    acquisition_type=AcquisitionType.INTEGRATION,
+                    averaging_mode=AveragingMode.CYCLIC,
+                ),
+            )
+            for qubit, ro_pulse in ro_pulses.items():
+                # average msr, phase, i and q over the number of shots defined in the runcard
+                r = results[ro_pulse.serial].serialize
+                r.update(
+                    {
+                        "wait[ns]": wait,
+                        "qubit_freqs[Hz]": qubits[qubit].drive_frequency,
+                        "qubit": qubit,
+                    }
                 )
+                data.add_data_from_dict(r)
+
+    else:
+        sweeper = Sweeper(
+            Parameter.delay,
+            waits,
+            [RX90_pulses1[qubit] for qubit in qubits],
```

Comment:
I believe this should be 
```suggestion
            [RX90_pulses2[qubit] for qubit in qubits],
```
We need to sweep the delay between the two drive pulses, so we sweep the start of the second pulse.

Reply:
I dont think so, for me this definition is way natural as the readout section can't be on its own delayed more than 2us and need to get linked to a drive or flux section with that delay. And also you will not delay the first pulse never with the other approach and that seemed odd to me.

Reply:
For me I just understand each pulse as two things.

1. Play the wave.
2. Delay for the next.

So the delay in this case should be part of the 1st pulse. That way when I get a sweeper I get both elements instead of having to check the last element of the sequence for the sweeper.values to apply the delay and the next to know the pulse on my loop. Maybe It's just the way I coded the driver but it seemed more straigtforward.

Reply:
Done